### PR TITLE
fix(ui): When feature flag `enable_uvicorn` is true require write lock when accessing Project.meltano attribute 

### PR DESCRIPTION
**Pull request by BuzzCutNorman**
This is an attempt at option 3 proposed in PR #6297 that asked to put the `write_lock()` behind the `enable_uvicorn` feature flag.

### CODE REVIEW

**Comment by aaronsteers** [comment]
@BuzzCutNorman - Very cool to see this PR created. I don't know if I'm the best approver so I won't jump into the code review quite yet. I do want to call out that this test is a new one and shouldn't have anything to do with the code you've changed. Please ignore:

image

**Comment by WillDaSilva** [comment]
@aaronsteers The dependency review workflow will work for forks once #6462 has been merged.

File: src/meltano/core/project.py [hunk 1]
```diff
@@ -231,9 +231,23 @@ def meltano(self) -> MeltanoFileHint:
...
         from .meltano_file import MeltanoFile
+        from .settings_service import FEATURE_FLAG_PREFIX, FeatureFlags
+        from .yaml import configure_yaml
 
-        with self._meltano_rw_lock.read_lock():
-            return MeltanoFile.parse(self.project_files.load())
+        with open(self.meltanofile) as melt_ff:
+            meltano_ff: dict = configure_yaml().load(melt_ff)
+
+        uvicron_enabled = (
```
Comment (cjohnhanson) [inline src/meltano/core/project.py:240]:
```suggestion
        uvicorn_enabled = (
```

File: src/meltano/core/project.py [hunk 2]
```diff
@@ -231,9 +231,23 @@ def meltano(self) -> MeltanoFileHint:
...
+            meltano_ff.get(f"{FEATURE_FLAG_PREFIX}.{FeatureFlags.ENABLE_UVICORN}")
+            or False
+        )
+
+        if uvicron_enabled:
```
Comment (cjohnhanson) [inline src/meltano/core/project.py:245]:
```suggestion
        if uvicorn_enabled:
```

**Review by cjohnhanson — CHANGES_REQUESTED** [review]
@BuzzCutNorman Thanks for tackling this! I've got a couple nitpicks but other than those, this looks good to go.

File: src/meltano/core/project.py (hunk 1 above)
Comment (BuzzCutNorman) [reply src/meltano/core/project.py]: @cjohnhanson thanks for catching that.  I made and committed the suggested changes.

**Review by cjohnhanson — APPROVED** [review]
Looks good!

We'll wait on an additional codeowner approval from either @pandemicsyn or @edgarrmondragon. Barring any requested changes there, this should be good to go.

**Review by edgarrmondragon — APPROVED** [review]

**Pull request merged**